tests/unit/configuration/test_inheritance_break_projects_and_groups.py

```diff
@@ -1,304 +1,221 @@
+                user1: developer
+                user2: developer
+    
+          "some_group/my_project":
+            secret_variables:
+              inherit: false
+              secret2:
+                key: bizz
+                value: buzz
         """
 
-    with pytest.raises(SystemExit) as exception:
-        Configuration(config_string=config_yaml).get_effective_config_for_project(
-            "another_group/another_project"
+        configuration = Configuration(config_string=config_yaml)
+
+        effective_config = configuration.get_effective_config_for_project(
+            "some_group/my_project"
         )
-    assert exception.type == SystemExit
-    assert exception.value.code == EXIT_INVALID_INPUT
-
-
-def test__get_effective_config_for_project__with_invalid_inheritance_break_set_at_group_level():
-    config_yaml = """
-    ---
-    projects_and_groups:
-      "some_group/*":
-        secret_variables:
-          inherit: false
-          first:
-            key: foo
-            value: bar
-
-      "some_group/my_project":
-        secret_variables:
-          second:
-            key: bizz
-            value: buzz
-    """
-
-    with pytest.raises(SystemExit) as exception:
-        Configuration(config_string=config_yaml).get_effective_config_for_project(
+
+        assert effective_config == {
+            "members": {
+                "users": {
+                    "user1": "developer",
+                    "user2": "developer",
+                }
+            },
+            "secret_variables": {
+                "secret2": {"key": "bizz", "value": "buzz"},
+            },
+        }
+
+    def test__inheritance_break__flag_set_at_group_level__group_inherits_nothing(
+        self,
+    ):
+        config_yaml = """
+        ---
+        projects_and_groups:
+          "*":
+            secret_variables:
+              secret1:
+                key: foo
+                value: bar
+
+          "some_group/*":
+            secret_variables:
+              inherit: false
+              secret2:
+                key: foo
+                value: bar
+        """
+
+        configuration = Configuration(config_string=config_yaml)
+
+        effective_config = configuration.get_effective_config_for_group("some_group")
+
+        secret_variables = effective_config["secret_variables"]
+
+        assert secret_variables == {
+            "secret2": {"key": "foo", "value": "bar"},
+        }
+
+    def test__inheritance_break__flag_set_at_group_level__group_is_highest_level_and_inherits_nothing(
+        self,
+    ):
+        config_yaml = """
+        ---
+        projects_and_groups:
+          "some_group/*":
+            secret_variables:
+              inherit: false
+              secret1:
+                key: foo
+                value: bar
+        """
+
+        configuration = Configuration(config_string=config_yaml)
+
+        effective_config = configuration.get_effective_config_for_group("some_group")
```

Comment:
> The "fix" is removing the check for group configs if the common config doesn't exist just to keep it simple for now. I did add one test case to ensure nothing is broken if the flag is set in the group and the common config doesn't exist.

That test is for the above comment I mentioned earlier.

Let's say we define a config for a group and subgroup with no common `*` config. The subgroup is trying to break the inheritance from the group. Since a subgroup is a group, the previous validation logic would throw an error because if the common config is empty and the subgroup contains `inherit: false` then it throws an error. However, it should be acceptable to have `inherit: false` in the subgroup because it's breaking the group-specific settings. Hopefully, that makes sense.

Reply:
Oh, that makes sense, thanks for the clarification!